src/Microsoft.AspNet.Http.Abstractions/Extensions/UseMiddlewareExtensions.cs

```diff
@@ -24,15 +24,28 @@ public static IApplicationBuilder UseMiddleware(this IApplicationBuilder builder
             {
                 var instance = ActivatorUtilities.CreateInstance(builder.ApplicationServices, middleware, new[] { next }.Concat(args).ToArray());
                 var methodinfo = middleware.GetMethod("Invoke", BindingFlags.Instance | BindingFlags.Public);
+                if (methodinfo == null)
+                {
+                    throw new InvalidOperationException("No public Invoke method found");
```

Comment:
Also, should have `Invoke` be passed in as a parameter to a call to format the message to ensure it doesn't localized by accident. Don't worry, we'll walk you through how to do this :smile: